protocol-units/da-sequencer/node/src/storage/mod.rs

```diff
@@ -188,7 +230,56 @@ impl DaSequencerStorage for Storage {
 	}
 
 	fn produce_next_block(&self) -> Result<Option<SequencerBlock>, DaSequencerError> {
-		todo!()
+		let cf_pending = self.db.cf_handle(cf::PENDING_TRANSACTIONS).ok_or_else(|| {
+			DaSequencerError::StorageAccess("Missing column family: pending_transactions".into())
+		})?;
+
+		let iter = self.db.iterator_cf(&cf_pending, rocksdb::IteratorMode::Start);
```

Comment:
The ordering of RockDb should be tested. In fact, I don't understand how RocksDb can order with `TxCompositeKey` that doesn't implement Ord. Or you suppose the save order is the right order. I can be ok because timestamp is set when the batch is received, so after the process is nearly sequential. If 2 batch switch, it means they have been received nearly at the same time. It works because there's only one Sequencer, so the order can be non-determinist. What do you think?
Last point, if it's the case, you can avoid saving the Tx with the timestamp and order in batch.

Reply:
Don't take into account I just see your other remarks.